File: app/src/main/java/info/firozansari/espressoexamples/TextSwapper.java
package info.firozansari.espressoexamples;

/**
 * Created by Firoz.
 */
public class TextSwapper {
    private String mBeforeText, mAfterText, mCurrentText;


    public TextSwapper(String beforeText, String afterText) {
        mBeforeText = beforeText;
        mAfterText = afterText;
        mCurrentText = mBeforeText;
    }

    public String swap() {
        mCurrentText = getCurrentText().equals(mBeforeText) ? mAfterText : mBeforeText;
        return mCurrentText;
    }

    public String getCurrentText() {
        return mCurrentText;
    }

}

File: app/src/main/java/info/firozansari/espressoexamples/TextSwapper.kt
package info.firozansari.espressoexamples

/**
 * Created by Firoz.
 */
class TextSwapper(private val mBeforeText: String, private val mAfterText: String) {
    var currentText: String
        private set

    fun swap(): String {
        currentText = if (currentText == mBeforeText) mAfterText else mBeforeText
        return currentText
    }

    init {
        currentText = mBeforeText
    }
}